apple/Sources/FerrostarCore/Location.swift

```diff
@@ -1,7 +1,9 @@
 import CoreLocation
+import UniFFI
 
 public protocol LocationProviding: AnyObject {
     var delegate: LocationManagingDelegate? { get set }
+    var authorizationStatus: CLAuthorizationStatus { get }
```

Comment:
Yeah, I don't like it either. 

That said, I can see it being very useful to just let the developer change and manipulate this value alongside their simulation. As far as whether it's a get set, I think we're fine leaving it's a get var since the UI doesn't technically need to manipulate this value. If someone does want that behavior, they can just cast their `LocationProviding` to whatever it is. However, in the `LiveLocationProvider` for example, I can see this being handled internally with any set permissions functionality as a get only.